app/services/db/sp_cost/sp_cost_summary_by_iaa.rb

```diff
@@ -0,0 +1,41 @@
+module Db
+  module SpCost
+    class SpCostSummaryByIaa
+      # @param [String] iaa
+      # @param [Range<Date>] iaa_range
+      def self.call(iaa:, iaa_range:)
+        params = {
+          iaa: iaa,
+          iaa_start_date: iaa_range.begin,
+          iaa_end_date: iaa_range.end,
+        }.transform_values { |v| ActiveRecord::Base.connection.quote(v) }
+
+        sql = format(<<~SQL, params)
+          SELECT
+            sp_costs.issuer
+          , sp_costs.ial
+          , sp_costs.cost_type
+          , MAX(service_providers.iaa) AS iaa
+          , MAX(service_providers.app_id) AS app_id
+          , COUNT(*)
+          FROM
+            sp_costs
+          INNER JOIN service_providers ON sp_costs.issuer = service_providers.issuer
+          WHERE
+            service_providers.iaa = %{iaa}
```

Comment:
It takes a long time to run each of these queries (they're a `created_at` index scan of the `sp_costs` table, but the table is *huge*).


Another version of the query could maybe just join `service_providers <-> sp_costs` and use the `service_providers.start_date` directly, but it was an even slower query